include/effectmanager.h

```diff
@@ -649,6 +649,15 @@ class EffectManager : public IJSONSerializable
         return _iCurrentEffect;
     }
 
+    size_t GetEffectIndexForID(const esp_uuid::UUID& id) const
+    {
+        for (size_t i = 0; i++; i < _vEffects.size())

```

Comment:
I'm not sure how any of this yields the index of the effect. Your first partial snippet doesn't show how to retrieve the index of the effect the for loop is at. The second snippet closes with a double reference to a variable `v` that doesn't exist outside of the `find_if()` lambda, and in that lambda uses a `struct_type` that doesn't exist at all.

In summary, it's not clear to me how to actually translate this to a better implementation of the current loop logic.

Maybe the code could be made more efficient by using a range for loop and keeping track of the index using a manually increased `i`. (That approach could then be simplified somewhat by using a range for loop initializer expression. However, that's a C++20 addition and this project uses C++17. So that's not an option at the moment.)
With the number of effects we're likely to look at, I doubt it would make much of a difference though.

In any case, we're now engaging in a discussion about the details of the implementation of one function. I opened this PR mainly to get feedback on the "UUID ID" approach at a more conceptual level. I'm curious if you have any input concerning that?